x/application/keeper/msg_server_delegate_to_gateway_test.go

```diff
@@ -177,8 +177,72 @@ func TestMsgServer_DelegateToGateway_FailGatewayNotStaked(t *testing.T) {
 
 	// Attempt to delegate the application to the unstaked gateway
 	_, err = srv.DelegateToGateway(wctx, delegateMsg)
-	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrAppGatewayNotFound)
 	foundApp, isAppFound := k.GetApplication(ctx, appAddr)
 	require.True(t, isAppFound)
 	require.Equal(t, 0, len(foundApp.DelegateeGatewayAddresses))
 }
+
+func TestMsgServer_DelegateToGateway_FailMaxReached(t *testing.T) {
+	k, ctx := keepertest.ApplicationKeeper(t)
+	srv := keeper.NewMsgServerImpl(*k)
+	wctx := sdk.WrapSDKContext(ctx)
+
+	// Generate an address for the application and gateway
+	appAddr := sample.AccAddress()
+	gatewayAddr := sample.AccAddress()
+	// Mock the gateway being staked via the staked gateway map
+	keepertest.StakedGatewayMap[gatewayAddr] = struct{}{}
```

Comment:
Please add a comment about `StakedGatewayMap` (where we define it) with a `// WARNING: Using this map may cause issues if running tests with parallelization > ``

Reply:
I will add parallisability in a seperate PR probably useful